### PR TITLE
chore(gen_id): using emqx_misc:gen_id/0, /1

### PR DESCRIPTION
**Pull request by tigercl**

### CODE REVIEW

**Review by terry-xiaoyu — APPROVED** [review]

File: apps/emqx_authz/src/emqx_authz_api_sources.erl [hunk 1]
```diff
@@ -452,21 +452,21 @@ write_cert(#{<<"ssl">> := #{<<"enable">> := true} = SSL} = Source) ->
     CertPath = filename:join([emqx:get_config([node, data_dir]), "certs"]),
     CaCert = case maps:is_key(<<"cacertfile">>, SSL) of
                  true ->
-                     {ok, CaCertFile} = write_file(filename:join([CertPath, "cacert-" ++ emqx_plugin_libs_id:gen() ++".pem"]),
+                     {ok, CaCertFile} = write_file(filename:join([CertPath, "cacert-" ++ emqx_misc:gen_id() ++".pem"]),
```
Comment (zmstone) [inline apps/emqx_authz/src/emqx_authz_api_sources.erl:455]: this not only makes the "certs" dir a chaos, but also grave yard for all old certificates.
it's hard to know which one is for which resource,
and hard to know which ones are still in use.

possible to add some correlation info in the file name ?

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl [hunk 2]
```diff
@@ -507,7 +507,7 @@ rule_id() ->
     gen_id("rule:", fun emqx_rule_registry:get_rule/1).
 
 gen_id(Prefix, TestFun) ->
-    Id = iolist_to_binary([Prefix, emqx_plugin_libs_id:gen()]),
+    Id = iolist_to_binary([Prefix, emqx_misc:gen_id()]),
     case TestFun(Id) of
         not_found -> Id;
         _Res -> gen_id(Prefix, TestFun)
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_engine.erl:513]: why deadloop here ?

Comment (zmstone) [reply apps/emqx_rule_engine/src/emqx_rule_engine.erl:513]: ah, never mind, forgot about the random part.

File: apps/emqx_authz/src/emqx_authz_api_sources.erl (hunk 1 above)
Comment (tigercl) [reply apps/emqx_authz/src/emqx_authz_api_sources.erl:455]: I originally used the ID of the instance as the name of an intermediate directory, but for some reason I deleted it.

Let me add it back, and I forgot to delete these certificate files when I delete the instance, I will implement these in another PR.

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl (hunk 2 above)
Comment (terry-xiaoyu) [reply apps/emqx_rule_engine/src/emqx_rule_engine.erl:513]: If the TestFun/1 returns something, then the id already exists. In this case we try again to create a new random id.

File: apps/emqx_authz/src/emqx_authz_api_sources.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_authz/src/emqx_authz_api_sources.erl:455]: ok, please remember to follow up with another PR

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**